vim-engine/src/main/kotlin/com/maddyhome/idea/vim/common/VimListenersNotifier.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2003-2023 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+package com.maddyhome.idea.vim.common
+
+import com.maddyhome.idea.vim.api.VimEditor
+import com.maddyhome.idea.vim.state.mode.Mode
+
+public class VimListenersNotifier {
+  public val modeChangeListeners: MutableList<ModeChangeListener> = mutableListOf()
+  public val editorFocusListeners: MutableList<EditorFocusListener> = mutableListOf()
+  public val macroRecordingListeners: MutableList<MacroRecordingListener> = mutableListOf()
```

Comment:
All these fields **must** be encapsulated. Only `add`/`remove` method should be available for the plugin writers. Otherwise, we may get a mess when the plugin writers will modify the listeners as they want.

Reply:
Why this one was marked as resolved?